### PR TITLE
fix(ui): add sandbox props to iframe

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add sandbox props to the iframe. Set undefined to allow all restrictions.
Also add the related issue for API response guard, inorder to remove the JSON response type assert. 

## Testing

test locally
@logto-io/eng

### CODE REVIEW

File: packages/ui/src/components/ConfirmModal/IframeConfirmModal.tsx [hunk 1]
```diff
@@ -38,6 +36,7 @@ const IframeConfirmModal = ({
         <div className={styles.content}>
           {isLoading && <LoadingIcon />}
           <iframe
+            sandbox={undefined}
```
Comment (gao-sun) [inline packages/ui/src/components/ConfirmModal/IframeConfirmModal.tsx:39]: should we set a proper value for this? i think `allow-scripts` would be fine?

Comment (simeng-li) [reply packages/ui/src/components/ConfirmModal/IframeConfirmModal.tsx:39]: as mentioned, setting undefined will enable all the restrictions including scripts not allowed. Should we intend to allow the scripts?

Comment (gao-sun) [reply packages/ui/src/components/ConfirmModal/IframeConfirmModal.tsx:39]: sry I misunderstood- noob me

**Review by gao-sun — APPROVED** [review]

**Review by orekihoutaro — APPROVED** [review]
LGMT!

**Pull request merged**